### PR TITLE
Refactor device-dashboard

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

File: examples/device-dashboard/Makefile [hunk 1]
```diff
@@ -20,27 +20,32 @@ ifeq ($(OS),Windows_NT)   # Windows settings. Assume MinGW compiler. To use VC:
   MAKE += WINDOWS=1 CC=$(CC)
 endif
 
-all: $(PROG)              # Default target. Build and run program
-	$(RUN) ./$(PROG) $(ARGS)
+# Default target. Build and run program
+all: $(PROG)
+	$(RUN) $(PROG) $(ARGS)
```
Comment (scaprile) [inline examples/device-dashboard/Makefile:25]: This works in Windows anyway, I guess make does the magic.
I like it this way better, it is more straightaway for the (Win)dozer

File: examples/device-dashboard/Makefile [hunk 2]
```diff
@@ -20,27 +20,32 @@ ifeq ($(OS),Windows_NT)   # Windows settings. Assume MinGW compiler. To use VC:
...
-clean:                    # Cleanup. Delete built program and all build artifacts
-	$(DELETE) $(PROG) *.o *.obj *.exe *.dSYM pack tmp mbedtls
+# Bundle JS libraries (preact, preact-router, ...) into a single file
+web_root/bundle.js:
+	curl -s https://npm.reversehttp.com/preact,preact/hooks,htm/preact,preact-router -o $@
 
-# see https://mongoose.ws/tutorials/tls/#how-to-build for TLS build options
+# Create optimised CSS. Prerequisite: npm -g i tailwindcss tailwindcss-font-inter
+web_root/main.css: web_root/index.html web_root/main.js
+	npx tailwindcss -o $@ --minify
```
Comment (scaprile) [inline examples/device-dashboard/Makefile:37]: This requires `npm` being installed, not standard

File: examples/device-dashboard/Makefile [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
+DELETE = rm -rf         # Command to remove files
+OUT ?= -o $(PROG)       # Compiler argument for output file
+SOURCES = main.c mongoose.c net.c       # Source code files
+CFLAGS = -W -Wall -Wextra -g -I.        # Build options
 
 # Mongoose build options. See https://mongoose.ws/documentation/#build-options
-CFLAGS_MONGOOSE += -DMG_ENABLE_PACKED_FS=1
+CFLAGS_MONGOOSE +=
 
 FILES_TO_EMBED ?= $(wildcard web_root/*)
```
Comment (scaprile) [inline examples/device-dashboard/Makefile:12]: I guess this macro is not used anymore and can be removed

File: examples/device-dashboard/main.c [hunk 4]
```diff
@@ -1,26 +1,31 @@
-// Copyright (c) 2020-2022 Cesanta Software Limited
+// Copyright (c) 2020-2023 Cesanta Software Limited
 // All rights reserved
 
-#include "mongoose.h"
+#include "net.h"
```
Comment (scaprile) [inline examples/device-dashboard/main.c:5]: I'd rather show including `mongoose.h` here, we are using Mongoose calls in this file and it is an example

File: examples/embedded-filesystem/packed_fs.c
Comment (scaprile) [inline examples/embedded-filesystem/packed_fs.c:1]: Nope, this example builds its own `packed_fs.c` with an About Cesanta page and shows gzipped vs non-gzipped differences

File: examples/device-dashboard/web_root/main.js
Comment (scaprile) [inline examples/device-dashboard/web_root/main.js:1]: Well, this one is totally different... no comments, I'll learn from this one

File: examples/device-dashboard/Makefile (hunk 2 above)
Comment (scaprile) [reply examples/device-dashboard/Makefile:37]: No more gzipping ?

Comment (cpq) [reply examples/device-dashboard/Makefile:37]: No more gzipping, that's too much IMO.
Yeah, npm needs to be installed - in the tutorial, we could just mention so.

File: examples/device-dashboard/Makefile (hunk 3 above)
Comment (cpq) [reply examples/device-dashboard/Makefile:12]: Yes! Can be removed

File: examples/device-dashboard/main.c (hunk 4 above)
Comment (cpq) [reply examples/device-dashboard/main.c:5]: Makes sense, yes

File: examples/device-dashboard/web_root/main.js
Comment (cpq) [reply examples/device-dashboard/web_root/main.js:1]: Yep. Totally different

**Pull request merged**